grouper/fe/util.py

```diff
@@ -189,6 +192,7 @@ def get_template_namespace(self):
             "perf_trace_uuid": self.perf_trace_uuid,
             "xsrf_form": self.xsrf_form_html,
             "alerts": self.get_alerts(),
+            "current_user_has_audit": self.current_user_has_audit,
```

Comment:
Is there a difference between calling these functions here and calling them from the template? I see `get_alerts` is the only one being called here.

Reply:
Passing it allows the code to only run if the result is of interest.  If I evaluate it, then every page will take a perf hit.

Reply:
It looks like this is going to be run on every page regardless. Also, each call to `user_has_permissions` does a database query. Should we concerned about this?

Reply:
Yeah, I think this is going to be pretty slow. It wouldn't be as bad if `user_permissions` was changed per the `FIXME` comment to use the graph instead, so I think that's a prerequisite.